ui/core/proto_utils/names.ts

```diff
@@ -212,6 +224,34 @@ export function getClassStatName(stat: Stat, playerClass: Class): string {
 	}
 }
 
+// TODO: Make sure BE exports the spell schools properly
+export enum SpellSchool {
+	None = 0,
+	Physical = 1 << (ProtoSpellSchool.SpellSchoolPhysical + 1),
+	Arcane = 1 << (ProtoSpellSchool.SpellSchoolArcane + 1),
+	Fire = 1 << (ProtoSpellSchool.SpellSchoolFire + 1),
+	Frost = 1 << (ProtoSpellSchool.SpellSchoolFrost + 1),
+	Holy = 1 << (ProtoSpellSchool.SpellSchoolHoly + 1),
+	Nature = 1 << (ProtoSpellSchool.SpellSchoolNature + 1),
+	Shadow = 1 << (ProtoSpellSchool.SpellSchoolShadow + 1),
+}
```

Comment:
I did it like this because in the end we should also refactor the schools in backend, since now they are defined in 2 places, 1 in proto for Encounter with incorrect values, and the correct one

Reply:
Well, the SpellSchool enum in the proto wasn't meant to encode all spell school combinations in the first place, just the values for the dropdown menu. I do agree that it should just have been an integer instead of the enum, since you really don't gain anything from using the enum; you just created more work for the backend to convert that again. And both should encode to 2 bytes in protobuf.

I guess this can be fixed now with the new versioning thingy if you write a migration for it, though I'm not sure if that's worth it.

But regardless, your code above here will break will break if this gets fixed which is why I suggested hardcoding it.

Reply:
Okay I understand. I've hardcoded them.